src/lib/viz/legend/LegendWidget.ts

```diff
@@ -0,0 +1,57 @@
+import { Layer } from '@/viz/layer';
+import { queryDOMElement } from '@/core/utils/dom';
+import { CartoError } from '@/core/errors/CartoError';
+
+export type LegendGeometryType = 'point' | 'line' | 'polygon';
+
+export type LegendProperties = {
```

Comment:
I think we should also use here the singular for the type, thus **LegendProperty**

Reply:
My mistake forget it, LegendProperties is the right choice!!